markup_tag_evaluation/awesome_align_tokenize.py

```diff
@@ -47,6 +58,16 @@ def write_lines(f: TextIOWrapper, texts: list[str]):
 
         src_sub = " ".join(token_src).replace(" ##", " ")
         tgt_sub = " ".join(token_tgt).replace(" ##", " ")
+        # Remove unks
+        src_unk_tokens = get_unk_tokens(src, tokenizer)
+        tgt_unk_tokens = get_unk_tokens(tgt, tokenizer)
+        for unk_tok in src_unk_tokens:
+            src_sub = src_sub.replace(UNK_TOKEN, unk_tok, 1)
+        for unk_tok in tgt_unk_tokens:
+            tgt_sub = tgt_sub.replace(UNK_TOKEN, unk_tok, 1)
+        assert UNK_TOKEN not in src_sub
+        assert UNK_TOKEN not in tgt_sub
```

Comment:
Maybe we could also assert that joining the final tokens again is equal to the original sentence, as a sanity check.

Reply:
Ah I guess that wouldn't work right away since we lose all the original whitespaces...